salt/_modules/metalk8s_drain.py

```diff
@@ -250,56 +223,14 @@ def get_controller(namespace, controller_ref):
           - None if not found
         Raises: CommandExecutionError if API fails
         '''
-        API_CONTROLLERS = {
-            "ReplicationController": {
-                "call": "CoreV1Api->list_namespaced_replication_controller",
-                "api_source": kubernetes.client.CoreV1Api,
-                "api_operation": "list_namespaced_replication_controller"
-            },
-            "DaemonSet": {
-                "call": "ExtensionsV1beta1Api->list_namespaced_daemon_set",
-                "api_source": kubernetes.client.ExtensionsV1beta1Api,
-                "api_operation": "list_namespaced_daemon_set"
-            },
-            "Job": {
-                "call": "CoreV1Api->list_namespaced_cron_job",
-                "api_source": kubernetes.client.BatchV1beta1Api,
-                "api_operation": "list_namespaced_cron_job"
-            },
-            "ReplicaSet": {
-                "call": "CoreV1Api->list_namespaced_replica_set",
-                "api_source": kubernetes.client.ExtensionsV1beta1Api,
-                "api_operation": "list_namespaced_replica_set"
-            },
-            "StatefulSet": {
-                "call": "AppsV1Api->list_namespaced_stateful_set",
-                "api_source": kubernetes.client.AppsV1Api,
-                "api_operation": "list_namespaced_stateful_set"
-            }
-        }
-
-        api_data = API_CONTROLLERS.get(controller_ref.kind)
-        if api_data is None:
-            raise CommandExecutionError(
-                "Unknown controller kind '{0}'".format(controller_ref.kind))
-
-        api_call = api_data["call"]
-        api_instance = api_data["api_source"]()
-
-        try:
-            return getattr(
-                api_instance,
-                api_data["api_operation"]
-            )(
-                namespace=namespace,
-                field_selector='metadata.name={0}'.format(controller_ref.name)
-            )
-        except (ApiException, HTTPError) as exc:
-            if isinstance(exc, ApiException) and exc.status == 404:
-                return None
-
-            log.exception('Exception when calling %s', api_call)
-            raise CommandExecutionError(str(exc))
+        return __salt__['metalk8s_kubernetes.get_object'](
+            name=controller_ref['name'],
+            kind=controller_ref['kind'],
+            apiVersion=controller_ref['api_version'],
+            namespace=namespace,
+            kubeconfig=self._kwargs.get('kubeconfig'),
+            context=self._kwargs.get('context')
```

Comment:
Why not just `**self._kwargs`?

Reply:
We could just thinking about the fact that `self._kwargs` are the `kwargs` of the drain so we maybe don't want to give them all to the sub-python-kubernetes calls

Reply:
Will be removed anyway with #1099